app/scripts/metamask-controller.js

```diff
@@ -903,11 +911,11 @@ export default class MetamaskController extends EventEmitter {
       getSnap: _getSnap,
       getSnapRpcHandler: _getSnapRpcHandler,
       getSnapState: _getSnapState,
-      showConfirmation: (origin, prompt, title, subtitle) =>
+      showConfirmation: (origin, confirmationData) =>
         this.approvalController.addAndShowApprovalRequest({
           origin,
           type: MESSAGE_TYPE.SNAP_CONFIRM,
-          requestData: { prompt, title, subtitle },
+          requestData: confirmationData,
```

Comment:
`confirmationData` is now an object of the form `{ prompt, description, textAreaContent }`.